infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -666,30 +672,43 @@ private void correctWrongEpochs(PollReport pollReport) {
             }
 
             // Update local layout copy.
-            serverContext.saveManagementLayout(latestLayout.get());
-            layout = serverContext.copyManagementLayout();
-
-            // In case of a partial seal, a set of servers can be sealed with a higher epoch.
-            // We should be able to detect this and bring the rest of the servers to this epoch.
-            long maxOutOfPhaseEpoch = Collections.max(wrongEpochs.values());
-            if (maxOutOfPhaseEpoch > layout.getEpoch()) {
-                layout.setEpoch(maxOutOfPhaseEpoch);
-            }
+            Layout newManagementLayout = latestLayout.get();
+            serverContext.saveManagementLayout(newManagementLayout);
```

Comment:
This does not guarantee that newManagementLayout is the latest layout.
Consider the following.
- We fetch the quorum layout - `latestLayout`. But we did not receive a response from the layout servers with the latest committed layout. Hence, `latestLayout` can be old.
- We save the management layout. This does not save this layout as it has already observed a newer layout.

This statement should be followed by something like this:
`newManagementLayout = serverContext.copyManagementLayout()`

This is an optimization to not waste the current detection round.
If we do not get the latest layout, the sealing will fail which will cause this poll round to fail.

Also, we are passing this to updateTrailingServers. Although the layout server has checks to discard commits with an old epoch, we do not want to attempt to commit an old layout.

Reply:
Let's discuss this offline and check that we don't break anything